torchmetrics/audio/pesq.py

```diff
@@ -0,0 +1,132 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Optional
+
+from torch import Tensor, tensor
+
+from torchmetrics.functional.audio.pesq import pesq
+from torchmetrics.metric import Metric
+from torchmetrics.utilities.imports import _PESQ_AVAILABLE
+
+
+class PESQ(Metric):
+    """PESQ (Perceptual Evaluation of Speech Quality)
+
+    This is a wrapper for the pesq package [1]. . Note that input will be moved to `cpu`
+    to perform the metric calculation.
+
+    .. note:: using this metrics requires you to have ``pesq`` install. Either install as ``pip install
+        torchmetrics[audio]`` or ``pip install pesq``
+
+    Forward accepts
+
+    - ``preds``: ``shape [...,time]``
+    - ``target``: ``shape [...,time]``
+
+    Args:
+        fs:
+            sampling frequency, should be 16000 or 8000 (Hz)
+        mode:
```

Comment:
Would this make more sense ?
```suggestion
        band_mode:
```

Reply:
band_mode=['wide', 'narrow']. 
Would you mind adding in the docstring what it means for the metric.

Reply:
I just borrow the name from `pesq` package,  it calls `mode`. so...

Reply:
see https://github.com/ludlows/python-pesq#usage-for-narrow-band-and-wide-band-modes